### PR TITLE
Specify, review and approve operation Interpolate-4

### PR DESCRIPTION
**Pull request by vgavrilo**
Specify, review and approve operation Interpolate-3

Ticket: CVS-29947

Description: addition of the support of the attribute `coordinate_transformation_mode` from ONNX specification of the ONNX layer Resize.

### CODE REVIEW

File: docs/ops/image/Interpolate_3.md [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+
+**Category**: Image processing
+
+**Short description**: *Interpolate* layer performs interpolation of independent slices in input tensor by specified dimensions and attributes.
+
+**Attributes**
+
+* *axes*
+
+  * **Description**: `axes` specify dimension indices where interpolation is applied, and `axes` is any unordered list of indeces of different dimensions of input tensor, e.g. `[0, 4]`, `[4, 0]`, `[4, 2, 1]`, `[1, 2, 3]`. These indeces should be non-negative integers from `0` to `rank(data) - 1` inclusively.  Other dimensions do not change. The order of elements in `axes` attribute matters, and mapped directly to elements in the 2nd input `target_spatial_shape`. Namely, `output_shape[axes[i]] = target_spatial_shape[i]` for all `i in range(0, len(axes))` and `output_shape[j] = input_shape[j] + pads_begin[j] + pads_end[j]` for `j not in axes`, `j in range(0, rank(data))`.
```
Comment (diyessi) [inline docs/ops/image/Interpolate_3.md:13]: `indices` not `indeces`

Comment (vgavrilo) [reply docs/ops/image/Interpolate_3.md]: Fixed.

**Comment by vladimir-paramuzov** [comment]
Shouldn't it be moved to opset4?

File: docs/ops/opset3.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ declared in `namespace opset3`.
 * [GroupConvolutionBackpropData](convolution/GroupConvolutionBackpropData_1.md)
 * [GRUCell](sequence/GRUCell_3.md)
 * [HardSigmoid](activation/HardSigmoid_1.md)
-* [Interpolate](image/Interpolate_1.md)
+* [Interpolate](image/Interpolate_3.md)
```
Comment (lazarevevgeny) [inline docs/ops/opset3.md:65]: As we discussed during the meeting we need to rename this operation to Intepolate_4 and move it to the docs/opst/opset4.md

Comment (vgavrilo) [reply docs/ops/opset3.md]: Fixed.

**Comment by vgavrilo** [comment]
> Shouldn't it be moved to opset4?

I agree. I have already done it.

File: docs/ops/image/Interpolate_4.md [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+
+**Category**: Image processing
+
+**Short description**: *Interpolate* layer performs interpolation of independent slices in input tensor by specified dimensions and attributes.
+
+**Attributes**
+
+* *axes*
+
+  * **Description**: `axes` specify dimension indices where interpolation is applied, and `axes` is any unordered list of indices of different dimensions of input tensor, e.g. `[0, 4]`, `[4, 0]`, `[4, 2, 1]`, `[1, 2, 3]`. These indices should be non-negative integers from `0` to `rank(data) - 1` inclusively.  Other dimensions do not change. The order of elements in `axes` attribute matters, and mapped directly to elements in the 2nd input `target_spatial_shape`. Namely, `output_shape[axes[i]] = target_spatial_shape[i]` for all `i in range(0, len(axes))` and `output_shape[j] = input_shape[j] + pads_begin[j] + pads_end[j]` for `j not in axes`, `j in range(0, rank(data))`.
```
Comment (lazarevevgeny) [inline docs/ops/image/Interpolate_4.md:13]: Make it an input.

File: docs/ops/image/Interpolate_4.md [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+  * **Description**: specifies type of interpolation
+  * **Range of values**: one of `nearest`, `linear`, `linear_onnx`, `cubic`, `area`
+  * **Type**: string
+  * **Default value**: none
+  * **Required**: *yes*
+
+* *coordinate_transformation_mode*
+
+  * **Description**: specifies how to transform the coordinate in the resized tensor to the coordinate in the original tensor
+  * **Range of values**: one of `half_pixel`, `pytorch_half_pixel`, `asymmetric`, `tf_half_pixel_for_nn`, `align_corners`
```
Comment (lazarevevgeny) [inline docs/ops/image/Interpolate_4.md:30]: Please, add verbal description of possible values.

File: docs/ops/image/Interpolate_4.md [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+
+* **Description**: *cube_coeff* specifies the parameter *a* for cubic interpolation (see, e.g.  [article](https://ieeexplore.ieee.org/document/1163711/)).  *cube_coeff* is used only when `mode == cubic`.
+  * **Range of values**: floating point number
+  * **Type**: any of supported floating point type
+  * **Default value**: `-0.75`
+  * **Required**: *no*
+
+**Inputs**
+
+*   **1**: `data` - Input tensor with data for interpolation. Type of elements is any supported type. Required.
```
Comment (lazarevevgeny) [inline docs/ops/image/Interpolate_4.md:79]: Floating point or i8 type.

File: docs/ops/image/Interpolate_4.md [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+
+*   **2**: `target_spatial_shape` - 1D tensor describing output shape for spatial axes. Number of elements matches the number of indices in *axes* attribute, the order matches as well. Required.
+
+**Outputs**
+
+*   **1**: Resulting interpolated tensor with elements of the same type as input `data` tensor. The shape of the output matches input `data` shape except spatial dimensions mentioned in `axes` attribute. For other dimensions shape matches sizes from `target_spaticl_shape` in order specified in `axes`.
```
Comment (lazarevevgeny) [inline docs/ops/image/Interpolate_4.md:85]: `target_spaticl_shape` -> `target_spatial_shape`

Comment (vgavrilo) [reply docs/ops/image/Interpolate_4.md]: Fixed.

File: docs/ops/image/Interpolate_4.md (hunk 5 above)
Comment (vgavrilo) [reply docs/ops/image/Interpolate_4.md]: Done.

File: docs/ops/image/Interpolate_4.md (hunk 3 above)
Comment (vgavrilo) [reply docs/ops/image/Interpolate_4.md]: Done.

File: docs/ops/image/Interpolate_4.md (hunk 4 above)
Comment (vgavrilo) [reply docs/ops/image/Interpolate_4.md]: Added.

File: docs/ops/image/Interpolate_4.md (hunk 3 above)
Comment (mitruska) [reply docs/ops/image/Interpolate_4.md]: ONNX [Resize ](https://github.com/onnx/onnx/blob/master/docs/Operators.md#resize) requires to specify scale or size for each of the data input dimensions, so interpolate `axes` is always like `0...(n-1)` where `n` is a number of elements in `target_spatial_shape` and equal to data rank. 
If `axes` is required input, we will have to add in onnx importer some operations like `shape_of` and `range` to provide the `axes`.
 
What do you think about making interpolate `axes` optional with `0...(n-1)` indices as default?

**Comment by mitruska** [comment]
ONNX [Resize ](https://github.com/onnx/onnx/blob/master/docs/Operators.md#resize) requires to specify scale or size for each of the data input dimensions, so interpolate `axes` is always like `0...(n-1)` where `n` is a number of elements in `target_spatial_shape` and equal to data rank. 
If `axes` is required input, we will have to add in onnx importer some operations like `shape_of` and `range` to provide the `axes`.
 
What do you think about making interpolate `axes` optional with `0...(n-1)` indices as default?

**Comment by vgavrilo** [comment]
> ONNX [Resize ](https://github.com/onnx/onnx/blob/master/docs/Operators.md#resize) requires to specify scale or size for each of the data input dimensions, so interpolate `axes` is always like `0...(n-1)` where `n` is a number of elements in `target_spatial_shape` and equal to data rank.
> If `axes` is required input, we will have to add in onnx importer some operations like `shape_of` and `range` to provide the `axes`.
> 
> What do you think about making interpolate `axes` optional with `0...(n-1)` indices as default?

I agree and I did it.

**Comment by rkazants** [comment]
@GlebKazantaev , @echuraev , please merge this PR. We've got required approvals.

**Comment by ilyachur** [comment]
Merged.

**Pull request merged**